src/import/import.rs

```diff
@@ -34,23 +34,110 @@ use super::{Config, Error, Result};
 const MAX_RETRY_TIMES: u64 = 5;
 const RETRY_INTERVAL_SECS: u64 = 3;
 
+/// Speed limit enforcement.
+///
+/// We need to restrict the maximum upload speed to avoid saturating the network
+/// bandwidth, which caused tikv-importer unable to contact PD.
+///
+/// This is implemented using the Token Bucket algorithm based on [1], but
+/// allowing bursting.
+///
+/// [1]: https://medium.com/smyte/rate-limiter-df3408325846
+struct SpeedLimiter {
+    // The maximum speed limit, unit = Byte/s.
+    speed_limit: f64,
+    // The last update time and the remaining amount of bytes can be sent during
+    // this second.
+    value: Mutex<(Instant, f64)>,
+}
+
+impl SpeedLimiter {
```

Comment:
Please add tests for the `SpeedLimiter`.

Reply:
Does TiKV have any existing utility to mock `now()` and `sleep()`? I'm going to create a new trait if not.